app/models.py

```diff
@@ -1,39 +1,38 @@
-from poplib import POP3_SSL_PORT
-from turtle import update
-from unittest import result
-from flask import session
 from werkzeug.security import generate_password_hash, check_password_hash
 
 from datetime import datetime
 
-from app import db
+from flask_login import UserMixin
+
+from app import db, login
+
 
 class BaseModel(db.Model):
+    __abstract__ = True
+
     id = db.Column(db.Integer, primary_key=True)
 
-    def find_by_id(self):
-        pass
+    @classmethod
+    def find_by_id(cls, _id):
+        return cls.query.filter_by(id=_id).first()
 
-    def find_by_name(self):
-        pass
-    
-    # TODO - Define __schema__ class
-    # TODO - Create base model and inheritt to all models
-    # id is a common field for every new model. You don't
-    # have to create it every time.
+    def save_to_db(self):
+        db.session.add(self)
+        db.session.commit()
 
+    def delete_from_db(self):
+        db.session.delete(self)
 
 
-class User(db.Model):
-    __tablename__ = 'user'
+class User(BaseModel, UserMixin):
+    __tablename__ = 'user_table'
 
-    id = db.Column(db.Integer, primary_key=True)
-    username = db.Column(db.String(64), index=True, unique=True)
-    email = db.Column(db.String(120), index=True, unique=True)
-    password_hash = db.Column(db.String(128))
+    username = db.Column(db.String(64), index=True, unique=True, nullable=False)
+    email = db.Column(db.String(120), index=True, unique=True, nullable=False)
+    password_hash = db.Column(db.String(128), nullable=False)
     created_at = db.Column(db.DateTime, index=True, default=datetime.utcnow)
-    updated_at = db.Column(db.DateTime, index=True)
-    user_posts = db.relationship('Post', backref='author', lazy='dynamic')
+    updated_at = db.Column(db.DateTime, default=db.func.now(), server_onupdate=db.func.now())
```

Comment:
Updated_at, create_at, timestamp fields can be updated automatically. We don't have to update them in routes or something.